var/spack/repos/builtin/packages/asio/package.py

```diff
@@ -32,6 +40,13 @@ class Asio(AutotoolsPackage):
     depends_on("automake", type="build")
     depends_on("m4", type="build")
     depends_on("libtool", type="build")
+    depends_on("pkgconfig", type="build", when="@1.23:")
+
+    conflicts(
+        "%gcc@12:",
+        when="@:1.22.0",
+        msg="asio v1.22.1 fixed missing includes necessary for gcc v12 and above",
+    )
```

Comment:
Do you have a link to an issue and/or a PR for this?

Reply:
I somewhat wanted to get around doing an issue for it... I tried to build `hpx` with gcc@13.1 when I ran into a build failure for `asio`, found https://github.com/chriskohlhoff/asio/issues/944 and the corresponding merged PR. It is also mentioned in the [1.22.1 release notes](https://think-async.com/Asio/asio-1.28.0/doc/asio/history.html#asio.history.asio_1_22_1) as

> - Added additional standard header file includes, as required by newer compilers.

Ishould have included that information above, but it was late yesterday... Should I create an issue?

Funnily, using the GitHub downloads saves us from another issue with `gcc@12:`: https://github.com/chriskohlhoff/asio/issues/1026

Reply:
Ah, no worries! https://github.com/chriskohlhoff/asio/pull/995 was exactly what I was hoping for. I wasn't looking for issues/workarounds/fixes for HPX or something else. Even having it listed here on the PR is already helpful, but maybe you can add it as a comment above the `conflicts`?

Reply:
For your convenience :wink: 
```suggestion
    # See https://github.com/chriskohlhoff/asio/issues/944 and
    # https://github.com/chriskohlhoff/asio/pull/995
    conflicts(
        "%gcc@12:",
        when="@:1.22.0",
        msg="asio v1.22.1 fixed missing includes necessary for gcc v12 and above",
    )
```